storage/badger/receipts.go

```diff
@@ -140,16 +140,29 @@ func (r *ExecutionReceipts) Index(blockID, receiptID flow.Identifier) error {
 		}
 
 		return nil
-	})
+	}
+	r.byBlockId = func(tx *badger.Txn, blockID flow.Identifier) ([]*flow.ExecutionReceipt, error) {
+		var receiptID flow.Identifier
+		err := operation.LookupExecutionReceipt(blockID, &receiptID)(tx)
+		if err != nil {
+			return nil, fmt.Errorf("could not lookup receipt ID: %w", err)
+		}
+		receipt, err := r.byID(receiptID)(tx)
+		if err != nil {
+			return nil, err
+		}
+		return []*flow.ExecutionReceipt{receipt}, nil
+	}
+	return r
 }
 
-func (r *ExecutionReceipts) IndexByExecutor(receipt *flow.ExecutionReceipt) error {
-	blockID := receipt.ExecutionResult.BlockID
-	executorID := receipt.ExecutorID
-	receiptID := receipt.ID()
-	return operation.RetryOnConflict(r.db.Update, func(tx *badger.Txn) error {
+func NewAllExecutionReceipts(collector module.CacheMetrics, db *badger.DB, results *ExecutionResults) *ExecutionReceipts {
```

Comment:
The demarcation between `NewExecutionReceipts` and `NewAllExecutionReceipts` is very tiny and may lead to confusion. Maybe renaming to `NewNonExecutionNodeReceipts` and `NewExecutionNodeReceipts` would help a more conveying name. The other option is to divide the interface `ExecutionReceipts` into `ExecutionNodeReceipts` and `NonExecutionNodeReceipts`. This is though just a suggestion for readability and maintenance. I leave it totally up to you. Definitely not a blocker.

Reply:
Agree, I was thinking something like:
`ExecutionReceiptsSingleIndex` `ExecutionReceiptsMutiIndex`
or 
`ExecutionNodeReceipts` `ConsensuNodeReceipts` (even though it's used by all nodes except execution). 
open to suggestions :)